src/rpc/rawtransaction.cpp

```diff
@@ -1091,6 +1093,20 @@ UniValue sendrawtransaction(const UniValue& params, bool fHelp)
     const CCoins* existingCoins = view.AccessCoins(hashTx);
     bool fHaveMempool = mempool.exists(hashTx);
     bool fHaveChain = existingCoins && existingCoins->nHeight < 1000000000;
+
+    // If we are configured to send transactions via an
+    // external service instead of broadcasting, do that
+    std::string strCmd = GetArg("-txsend", "");
+    if (!strCmd.empty()) {
+        if (fHaveChain) {
+            throw JSONRPCError(RPC_TRANSACTION_ALREADY_IN_CHAIN, "transaction already in block chain");
+        }
+        boost::replace_all(strCmd, "%s", EncodeHexTx(tx));
+        boost::thread t(runCommand, strCmd); // thread runs free
```

Comment:
Please factor out these into a function so that this code isn't duplicated in `wallet.cpp`. (Yes I know it's only two lines but it's not *entirely* trivial.)